integration_tests/simple_hover.cpp

```diff
@@ -13,31 +33,55 @@ TEST_F(SitlTest, ActionSimpleHover)
 
     // Wait for device to connect via heartbeat.
     std::this_thread::sleep_for(std::chrono::seconds(2));
+    ASSERT_TRUE(dc.is_connected());
 
     Device &device = dc.device();
 
-    while (!device.telemetry().health_all_ok()) {
+    for (int i = 0; ; ++i) {
```

Comment:
Just FMI, why not a while loop? Seem odd to see an infinite for loop :-)

Reply:
Ignore that, I see you want to count out 10 seconds.

Reply:
Yer it's a bit ugly but I haven't found a prettier way yet.

An alternative is to use the async interface and then do a `wait_for` on the promise.

Reply:
Not prettier, but perhaps a little more intuitive
```
int itter=0;
while (!device.telemetry().health_all_ok()) {
   std::cout << "waiting for device to be ready" << std::endl;
   std::this_thread::sleep_for(std::chrono::seconds(1));
   ASSERT_LT(++itter, 10);
}
```

Reply:
Yer, you're right. That's nicer. I'll change it.